Source/View Controllers/UploadViewController.swift

```diff
@@ -342,15 +361,15 @@ class UploadViewController: NSViewController {
         if saveCredentials {
             if credentialsAvailable {
                 do {
-                    try SecurityWrapper.saveCredentials(username: username,
-                                                        password: password,
+                    try SecurityWrapper.saveCredentials(username: username!,
+                                                        password: password!,
```

Comment:
Could we put a `guard let username = username` and for the `guard let password = password` here instead of force-unwrapping?

I'm not sure what to do here if username or password is nil. We could either return without reacting like on line 372, or throw up an alert like we do in the `checkConnectionPressed` method. Thoughts?

Reply:
Thought about this, but as the check of `if credentialsAvailable {` is done before which will do the `nil` check, I thought this was the cleaner way.

Reply:
We don't need `guard let` but adding the unwrapping as part of an `if let` along with the `credentialsAvailable` check just above the `do {` line I think makes sense and is pretty easy to read.

```
if let username = username, let password = password, credentialsAvailable {
```